lib/engine/game/g_1862/game.rb

```diff
@@ -623,6 +626,94 @@ def add_obligation(entity, corporation)
           @chartered[corporation] = true
         end
 
+        # called at end of stock round
+        def enforce_obligations
+          @players.each do |player|
+            remaining_fine = 0
+            player.companies.each do |company|
```

Comment:
dup since you're mutating this?

Reply:
Yep. I missed this because my test case only had one missed obligation. Two missed obligations on a player would be quite the dumpster fire, but theoretically possible.